### PR TITLE
docs: Cleanup Windows build instructions

### PR DESCRIPTION
**Pull request by juan-lunarg**
Minimize docs as much as possible so they don't become out of date.

Refer to CMake docs which cover -A / -G usage.

### CODE REVIEW

File: BUILD.md [hunk 1]
```diff
@@ -179,57 +179,29 @@ cmake -S . -B build/ --preset dev
...
+# Open the Visual Studio solution
+cmake --open build
 ```
 
-Note: VS2019 and higher target "x64" by default. VS2017 and lower target "Win32" by default
-
 Check the [official CMake documentation](https://cmake.org/cmake/help/latest/manual/cmake-generators.7.html#visual-studio-generators) for further information on Visual Studio generators.
 
-The above steps create a Windows solution file named
-`Vulkan-ValidationLayers.sln` in the build directory.
```
Comment (juan-lunarg) [inline BUILD.md:220]: CMake no longer generates a solution named `Vulkan-ValidationLayers.sln`.

**Review by juan-lunarg — COMMENTED** [review]
Minimize docs as much as possible so they don't become out of date.

Refer to CMake docs which cover -A / -G usage.

**Review by artem-lunarg — APPROVED** [review]
🪟👍

File: BUILD.md [hunk 2]
```diff
@@ -179,57 +179,29 @@ cmake -S . -B build/ --preset dev
...
 - Git Client Support
-  - [Git for Windows](http://git-scm.com/download/win) is a popular solution
-    for Windows
-  - Some IDEs (e.g., [Visual Studio](https://www.visualstudio.com/),
-    [GitHub Desktop](https://desktop.github.com/)) have integrated
-    Git client support
+  - [Git for Windows](http://git-scm.com/download/win) is a popular solution for Windows
 
-### Windows Build - Microsoft Visual Studio
+NOTE: [Visual Studio 17 2022](https://cmake.org/cmake/help/latest/generator/Visual%20Studio%2017%202022.html) requires need at least CMake `3.21`
```
Comment (arno-lunarg) [inline BUILD.md:189]:
```suggestion
NOTE: [Visual Studio 17 2022](https://cmake.org/cmake/help/latest/generator/Visual%20Studio%2017%202022.html) requires at least CMake `3.21`
```

**Review by arno-lunarg — APPROVED** [review]
LGTM! One typo

**Pull request merged**